include/seqan3/range/container/constexpr_string.hpp

```diff
@@ -0,0 +1,430 @@
+    {
+        return &_lit[0];
+    }
+
+    /*!\brief Returns the begin to the string.
+     *
+     * \returns `const_iterator` An iterator to the begin of the string.
+     * \par Exceptions
+     * No-throw guarantee.
+     */
+    constexpr const_iterator cbegin() const noexcept
+    {
+        return &_lit[0];
+    }
+
+    /*!\brief Returns the end to the string.
+     *
+     * \returns `iterator` An iterator to the end of the string.
+     * \par Exceptions
+     * No-throw guarantee.
+     */
+    constexpr iterator end() noexcept
+    {
+        return &_lit[N];
+    }
+
+    /*!\brief Returns the end to the string.
+     *
+     * \returns `iterator` An iterator to the end of the string.
+     * \par Exceptions
+     * No-throw guarantee.
+     */
+    constexpr const_iterator end() const noexcept
+    {
+        return &_lit[N];
+    }
+
+    /*!\brief Returns the end to the string.
+     *
+     * \returns `const_iterator` An iterator to the end of the string.
+     * \par Exceptions
+     * No-throw guarantee.
+     */
+    constexpr const_iterator cend() const noexcept
+    {
+        return &_lit[N];
+    }
+
+    /*!\brief Compares two strings for equality.
+     *
+     * \returns `bool` `true` if both container are equal, `false` otherwise.
+     * \par Exceptions
+     * No-throw guarantee.
+     */
+    template <std::size_t N2>
+    constexpr bool operator==(constexpr_string<N2> const & rhs) const noexcept
+    {
+        if constexpr (N != N2)
+            return false;
+
+        const_iterator it_rhs = rhs.cbegin();
+        for (const_iterator it = cbegin(); it != cend(); ++it, ++it_rhs)
+        {
+            if (*it != *it_rhs)
+                return false;
+        }
+
+        return true;
+    }
+
+    /*!\brief Compares two strings for inequality.
+     *
+     * \returns `bool` `false` if both container are equal, `true` otherwise.
+     */
+    template <std::size_t N2>
+    constexpr bool operator!=(constexpr_string<N2> const & rhs) const noexcept
+    {
+        return !(*this == rhs);
+    }
+
+    /*!\brief Swaps two strings.
+     * Note both strings must have the same size in order to swap them.
+     */
+    constexpr void swap(constexpr_string & other) noexcept
+    {
+        _lit.swap(other._lit);
+    }
+
+    /*!\brief Returns the content represented as std::string.
+     *
+     * \returns \c std::string The stored string.
+     * \par Exceptions
+     * Strong exception guarantee. No data is modified.
+     */
+    std::string string() const
+    {
+        return std::string{ _lit.cbegin(), _lit.cend() - 1};
+    }
+
+    /*!\brief Returns the content represented as 0-terminated c-style string.
+     *
+     * \returns `const char *` The stored string.
+     * \par Exceptions
+     * No-throw guarantee.
+     */
+    constexpr const char * c_str() const noexcept
+    {
+        return _lit.data();
+    }
+    //!\}
+
+    //!\brief Implicit string conversion.
+    operator std::string() const
```

Comment:
Do we want implicit conversion, is explicit not enough?

Reply:
probably not